src/Functions/MatchImpl.h

```diff
@@ -63,14 +63,31 @@ inline bool likePatternIsSubstring(std::string_view pattern, String & res)
 
 }
 
-/** 'like'             - if true, treat pattern as SQL LIKE, otherwise as re2 regexp.
- *  'negate'           - if true, negate result
- *  'case_insensitive' - if true, match case insensitively
- *
-  * NOTE: We want to run regexp search for whole columns by one call (as implemented in function 'position')
-  *  but for that, regexp engine must support \0 bytes and their interpretation as string boundaries.
-  */
-template <typename Name, bool like, bool negate, bool case_insensitive>
+// Below enums make instantiations of MatchImpl<> more readable
+
+enum class MatchSyntax
+{
+    Like,
+    Re2
+};
+
+enum class NegateMatch
+{
+    No,
+    Yes
+};
+
+enum class MatchCaseInsensitively
```

Comment:
Why don't go further and name it 

CaseSensitiveness
- Sensitive
- Insensitive